assets/components/subscriptionsByCountryGroup/subscriptionsByCountryGroup.jsx

```diff
@@ -5,64 +5,330 @@
 import { type CommonState } from 'helpers/page/page';
+import { displayPrice } from 'helpers/subscriptions';
+import { addQueryParamsToURL } from 'helpers/url';
+import { getCampaign, type ReferrerAcquisitionData } from 'helpers/tracking/acquisitions';
+import {
+  getSubsLinks,
+  type SubsUrls,
+  iOSAppUrl,
+  androidAppUrl,
+  dailyEditionUrl,
+} from 'helpers/externalLinks';
+import { classNameWithModifiers } from 'helpers/utilities';
+import { countryGroups } from 'helpers/internationalisation/countryGroup';
+import { type ImageId } from 'helpers/theGrid';
 
 
 // ----- Types and State Mapping ----- //
 
 type PropTypes = {
   countryGroupId: CountryGroupId,
   headingSize: HeadingSize,
+  referrerAcquisitionData: ReferrerAcquisitionData,
 };
 
-
 function mapStateToProps(state: { common: CommonState }) {
 
   return {
     countryGroupId: state.common.internationalisation.countryGroupId,
+    referrerAcquisitionData: state.common.referrerAcquisitionData,
   };
 
 }
 
 
+// ----- Setup ----- //
+
+const appReferrer = 'utm_source=support.theguardian.com&utm_medium=subscribe_landing_page';
+const internationalAppReferrer = 'utm_source=support.theguardian.com&utm_medium=subscribe_landing_page&utm_campaign=international_subs_landing_pages';
+
+
 // ----- Component ----- //
 
 function SubscriptionsByCountryGroup(props: PropTypes) {
 
-  const { countryGroupId, headingSize, ...otherProps } = props;
+  const {
+    countryGroupId, headingSize, referrerAcquisitionData, ...otherProps
+  } = props;
+  const subsLinks = getSubsLinks(
+    countryGroupId,
+    referrerAcquisitionData.campaignCode,
+    getCampaign(referrerAcquisitionData),
+    referrerAcquisitionData,
+  );
+  const className = classNameWithModifiers(
+    'component-subscriptions-by-country-group',
+    [countryGroups[countryGroupId].supportInternationalisationId],
+  );
 
   if (countryGroupId === 'GBPCountries') {
     return (
-      <div {...otherProps}>
-        <DigitalSubscriptionsContainer
-          headingSize={headingSize}
-        />
-        <PaperSubscriptionsContainer
-          headingSize={headingSize}
-        />
+      <div className={className} {...otherProps}>
+        <Digital headingSize={headingSize} subsLinks={subsLinks} countryGroupId={countryGroupId} />
+        <Paper headingSize={headingSize} subsLinks={subsLinks} countryGroupId={countryGroupId} />
       </div>
     );
   }
 
   return (
-    <div {...otherProps}>
-      <InternationalSubscriptions
-        countryGroupId={countryGroupId}
-        headingSize={headingSize}
-        {...otherProps}
-      />
+    <div className={className} {...otherProps}>
+      <International headingSize={headingSize} subsLinks={subsLinks} />
     </div>
   );
 
 }
 
 
+// ----- Auxiliary Components ----- //
+
+function Digital(props: {
+  headingSize: HeadingSize,
+  subsLinks: SubsUrls,
+  countryGroupId: CountryGroupId,
+}) {
+  return (
+    <ThreeSubscriptions heading="Digital Subscriptions">
```

Comment:
Might it be better to call this something like `SubscriptionBlock` or `SubscriptionGroup`? Having the name include the number of items risks making it wrong if we decide to group them differently.

Reply:
The `children` type is [a tuple](https://flow.org/en/docs/types/tuples/) that means exactly three elements have to be passed. Does that cover this problem?

Reply:
Not really, it's more about if we decide from a design point of view that it would look nicer to have four products in a group or two with a 'hero' product above, then we can still use this component to group them but the name is wrong. 

In that case we'd need to change the tuple so we could also change the name at that point I guess, but I'm not sure what value it adds having Three in the name anyway, so we could avoid the issue from the start.

Reply:
I think in that situation we'd have to either change this component to use that new layout, or create a new one. Most of the value in this component is the CSS, which is tied to the 3-way layout and hence why I thought it was important to make that explicit in the name. Without knowing in advance what a new layout would look like it's hard to make that generic, so I think it's a change we'd make at the point when we have a design that pushes us to do it?